### PR TITLE
Replace From trait with TryFrom trait for datafusion-proto crate #3401

### LINKED ISSUE #3401: Replace From trait with TryFrom trait for datafusion-proto crate
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
Replace From trait with TryFrom trait for ArrowTypeEnum. According to https://github.com/apache/arrow-datafusion/pull/3365#discussion_r964066994

**Describe the solution you'd like**
Replace From trait with TryFrom trait for ArrowTypeEnum

**Describe alternatives you've considered**
Leave as is

**Additional context**
None

### PR DESCRIPTION
**Pull request by comphead**
# Which issue does this PR close?

Closes #3401 .

 # Rationale for this change

Replacing From with TryFrom will help with error handling

# What changes are included in this PR?

Replace From trait with TryFrom trait for datafusion-proto crate #3401

# Are there any user-facing changes?

No

### CODE REVIEW

**Comment by comphead** [comment]
@andygrove this is related to your comment https://github.com/apache/arrow-datafusion/pull/3365#discussion_r964066994

Please review

File: datafusion/proto/src/logical_plan.rs [hunk 1]
```diff
@@ -719,7 +719,7 @@ impl AsLogicalPlan for LogicalPlanNode {
                         })
                     }
                 };
-                let schema: protobuf::Schema = schema.as_ref().into();
+                let schema: protobuf::Schema = schema.as_ref().try_into()?;
```
Comment (alamb) [inline datafusion/proto/src/logical_plan.rs:722]: 👍

**Review by alamb — APPROVED** [review]
Looks great to me -- thanks @comphead

**Review by andygrove — APPROVED** [review]
Thank you @comphead!

**Pull request merged**